browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchAdapter.java

```diff
@@ -0,0 +1,109 @@
+package com.braintreepayments.browserswitch;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+
+import androidx.annotation.VisibleForTesting;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+
+public class BrowserSwitchAdapter implements IBrowserSwitchClient {
+
+    static BrowserSwitchAdapter newInstance() {
+        return newInstance(
+                BrowserSwitchClient.newInstance(), BrowserSwitchConfig.newInstance());
+    }
+
+    @VisibleForTesting
+    static BrowserSwitchAdapter newInstance(BrowserSwitchClient internalClient, BrowserSwitchConfig config) {
+        return new BrowserSwitchAdapter(internalClient, config);
+    }
+
+    private BrowserSwitchConfig config;
+    private BrowserSwitchClient internalClient;
+
+    private BrowserSwitchAdapter(BrowserSwitchClient internalClient, BrowserSwitchConfig config) {
+        this.internalClient = internalClient;
```

Comment:
What was the motivation for creating `BrowserSwitchAdapter` instead of just `BrowserSwitchClient`?